### PR TITLE
biting the webpack bullet

### PR DESCRIPTION
**Pull request by gingerlime**
* removed browserify and moved to webpack
* upgraded libraries in package.json to latest versions
* updated npm run commands to use webpack instead of browserify
* upgraded coffeescript to v2 and updated code
* adapted tests to latest version of sinon
* moved from travis to circleci

### CODE REVIEW

**Comment by gingerlime** [comment]
@joker-777 would be great if you could take a look when you a chance ... I moved things over to webpack from browserify (can't say I'm that excited about it, but it seems like the sane choice these days). It can now also get `require`d more easily with `require("alephbet")`...

File: webpack.config.coffee [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+file_name = (mode) ->
+  if mode == "production"
+    "alephbet.min.js"
+  else
+    "alephbet.js"
+
+module.exports = (env, argv) ->
+  target: "web"
+  node:
+    fs: "empty"
```
Comment (joker-777) [inline webpack.config.coffee]: Why did you add this?

File: webpack.config.coffee [hunk 2]
```diff
@@ -0,0 +1,43 @@
+path = require("path")
+
+file_name = (mode) ->
+  if mode == "production"
+    "alephbet.min.js"
+  else
+    "alephbet.js"
+
+module.exports = (env, argv) ->
+  target: "web"
```
Comment (joker-777) [inline webpack.config.coffee]: You could leave this out since it is the default. Just wanted to mention it.

File: webpack.config.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      ".js"
+    ]
+  module:
+    rules: [
+      test: /\.coffee$/
+      use: [
+        loader: "coffee-loader"
+        options:
+          transpile:
+            plugins: [ "@babel/plugin-syntax-dynamic-import" ]
```
Comment (joker-777) [inline webpack.config.coffee]: Do you use dynamic imports?

**Comment by joker-777** [comment]
@gingerlime Added a few comments but there are very minor. In general it looks good.

File: webpack.config.coffee (hunk 1 above)
Comment (gingerlime) [reply webpack.config.coffee]: Good point :) probably unnecessary... I did a fair bit of trial and error by copying stuff from other places. I can try to remove it and hopefully it would still work.

File: webpack.config.coffee (hunk 2 above)
Comment (gingerlime) [reply webpack.config.coffee]: Good to know. I usually prefer to keep things explicit unless they are really obvious/go against the convention...

File: webpack.config.coffee (hunk 3 above)
Comment (gingerlime) [reply webpack.config.coffee]: I "borrowed" this from Kenhub... I guess I don't, so this can be removed as well.

**Comment by gingerlime** [comment]
Thanks @joker-777 very useful comments!

File: webpack.config.coffee (hunk 1 above)
Comment (joker-777) [reply webpack.config.coffee]: I didn't know what it means so I had to look at the documentation. I don't think you need it.

File: webpack.config.coffee (hunk 3 above)
Comment (joker-777) [reply webpack.config.coffee]: Yes, I think you can remove it and also the package you installed for it.

File: webpack.config.coffee (hunk 1 above)
Comment (gingerlime) [reply webpack.config.coffee]: It seems to be needed for webpack.test.config.coffee otherwise the tests fail
